src/mscorlib/shared/System/Numerics/Vector.cs

```diff
@@ -763,6 +767,36 @@ private Vector(ref Register existingRegister)
         {
             this.register = existingRegister;
         }
+
+        /// <summary>
+        /// Constructs a vector from the given span. The span must contain at least Vector'T.Count elements.
+        /// </summary>
+        [Intrinsic]
+        public unsafe Vector(Span<T> values)
+            : this()
+        {
+            if ((typeof(T) == typeof(Byte))
+                || (typeof(T) == typeof(SByte))
+                || (typeof(T) == typeof(UInt16))
+                || (typeof(T) == typeof(Int16))
+                || (typeof(T) == typeof(UInt32))
+                || (typeof(T) == typeof(Int32))
+                || (typeof(T) == typeof(UInt64))
+                || (typeof(T) == typeof(Int64))
+                || (typeof(T) == typeof(Single))
+                || (typeof(T) == typeof(Double)))
+            {
+                if (values.Length < Count)
+                {
+                    throw new IndexOutOfRangeException(SR.Format(SR.Arg_InsufficientNumberOfElements, Vector<T>.Count, nameof(values)));
```

Comment:
It may help users if this exception told them `values.Length` as well.

Reply:
I had thought about this and was ok for this instance. But there is another instance for `T[]` based constructor which I reintroduced just now as per review [this comment](https://github.com/dotnet/coreclr/commit/357e2133199f890f2a4987da85dceeb5d72ecbc8#r27916246) by @4creators. In that case, the statement becomes complex when an array and an index is mentioned. In that case it is not simply the number of elements in `value` but the relative number of elements in `value` with respect to the index.

I was trying to have consistency in the exception being thrown across the three usages and with few other exceptions in the resources file e.g.
* Arg_ElementsInSourceIsGreaterThanDestination - Number of elements in source vector is greater than the destination array
* Argument_InsufficientSpaceToCopyCollection - The specified space is not sufficient to copy the elements from this Collection.

And anyways, the user would know the number of elements that (s)he is sending...